pkg/apis/options/proxy.go

```diff
@@ -0,0 +1,38 @@
+package options
+
+type ProxyOptions struct {
```

Comment:
I would like to separate this struct into multiple sections. Please free to give some input

Reply:
@JoelSpeed @kvanzuijlen @braunsonm 

Any ideas on how to better subdivide this struct?

I already tried to identify some categories of flags indicated by the comments `networking`, `authentication`, `proxy` maybe something like `routing`, `security` would be good as well. I would like to get rid of the `ProxyOptions` struct altogether and instead have multiple structs for better isolation of concerns.

Reply:
I would prefer the `routing` and `security` split instead of what you have here right now.

I would caution some of the organization here. Is `SkipJwtBearerTokens` `proxy` related whereas `TrustedIPs` is in the `networking` section? These seem misplaced to me and I think just `routing` and `security` might be more clear.

Reply:
The initial section names I wrote in the code comments were done on a whim as a starting point for me. While I wrote the request to you guys, I came up with the `routing` and `security` split which is indeed a better solution in my opinion. Nice to hear that you agree.

Reply:
What do you guys think about the new grouping? If it is fine with everybody I would create three structs and finalize the PR.

Reply:
I think this is fine. However the difference between `authentication` and `security` might not be very clear to users if these parameters are going to be grouped differently in the YAML configuration.